cauldron/redis_cache.py

```diff
@@ -152,6 +152,13 @@ def hmset(cls, field, value, namespace=''):
         with (yield from cls.get_pool()) as redis:
             yield from redis.hmset(namespace, field, value)
 
+    @classmethod
+    @coroutine
+    def hmset_with_ttl(cls,field,value,ttl,namespace =''):
+        with (yield from cls.get_pool()) as redis:
+            yield from redis.hmset(namespace, field, value)
+            redis.expire(namespace,ttl)
```

Comment:
This will reset the expiry to ttl seconds each time this method is called. So if within "ttl" seconds this method is called again and again the key will never expire. But in our use case key must expire after "ttl" seconds from the time it was first created.

Also, need to check if expire() is a coroutine or not. If it's a coroutine "yield from" is missing here.